AppController/djinn.rb

```diff
@@ -1014,7 +1013,11 @@ def set_parameters(layout, options, secret)
 
     @nodes.each { |node|
       if node.jobs.include? 'compute'
-        @options['compute_instance_type'] = node.instance_type
+        if not node.instance_type.nil?
```

Comment:
maybe simplify

```
if node.instance_type.nil?
  @options['compute_instance_type'] = @options['instance_type']
else
  @options['compute_instance_type'] = node.instance_type
end

break
```

also what happens if someone's ASF is:
```
ips_layout:
  -
    roles: [master, database, compute]
    nodes: 1
    instance_type: m3.xlarge
  -
    roles: [compute]
    nodes: 3
    instance_type: m1.xlarge
```

Reply:
Sure I can simplify that piece of code. And for multiple compute nodes specified with different instance types, this PR now will pick up randomly the instance type for a compute node it encounters. It will be difficult to maintain a list of compute nodes and their instance types and make the decision of which instance type to use for autoscaling. I would vote for allowing an option for the user to specify a particular instance type to autoscale with if needed so that the decision is not random. But currently, I did not see any harm to assume the instance type to autoscale with as long as it is specified for any one of the compute nodes.